app/ui/browser-modern/views/browser/tabbar/tab.jsx

```diff
@@ -81,38 +85,35 @@ class Tab extends Component {
 
   handleTabPick = e => {
     if (e.button === 1) {
-      this.handleTabClose(e);
+      this.props.dispatch(PageEffects.destroyPageSession(this.props.pageId));
     } else {
       this.props.dispatch(PageActions.setSelectedPage(this.props.pageId));
     }
   }
 
-  handleTabClose = e => {
-    this.props.dispatch(PageEffects.destroyPageSession(this.props.pageId));
-    e.stopPropagation();
+  handleTabDoubleClick = () => {
+    this.props.dispatch((dispatch, getState) => {
```

Comment:
Why do we need to get the state and run the selector here?  `this.props.isPinned` should be up to date here, right?

So we could do this instead:

    if (this.props.isPinned) {
      this.props.dispatch(PageActions.setPagePinned(this.props.pageId));
    } else {
      this.props.dispatch(PageActions.setPageUnpinned(this.props.pageId));
    }